ADD: Multisig seed with passphrase

limpbrains opened the pull request:
Available in Advanced mode

this PR also improves Electrum desktop MS wallet import. You can now import electrum seed wallet with passphrase or with xpriv and sign a tx with it.

https://user-images.githubusercontent.com/155891/148687191-9122bd23-a920-4fbb-85ac-85b756ff39d1.mp4

closes #3665

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/7pffuy5whw1ynjj9gr77kextqc?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9335)

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/f9eu1pjwu6yyye8hemu6qypwj0?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9336)

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/3a9w29k5r1h8kcw8k2zzkzg4wc?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9341)

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/k6ewjw541d83bvjb7nvz9xjk98?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9371)

limpbrains commented:
`viewEditMultisigCosigners` needs to be updated as well

limpbrains commented:
In last commit I removed `wallet.replaceCosigner` method and added `replaceCosignerSeedWithXpub` and `replaceCosignerXpubWithSeed`. They are simpler, do all logic in place, doesn't change rearrange cosigners. This fix #4240

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/ce0uq5azxvfaagcbefz609e768?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9376)

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/gnnaqquh0264dtke4dxpjck0rw?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9377)

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/8qkmxdnxbme14dmw4vh6qvefr4?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/9406)

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @marcosrdz] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

limpbrains commented:
@Overtorment @marcosrdz let's merge it. Or do you have any concerns?

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

DrockDaniel commented:
Sorry I'm just recovering from COVID-19 just chilling here. What's a PR?

On Sun, Jan 23, 2022, 4:19 AM GLaDOS ***@***.***> wrote:

> Wake the fuck up samurai, we have PRs to merge
>
> [image: image]
> <https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png>
>
> [all PRs for @marcosrdz <https://github.com/marcosrdz>]
> https://github.com/BlueWallet/BlueWallet/pulls/review-requested/marcosrdz
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/BlueWallet/BlueWallet/pull/4350#issuecomment-1019453854>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUOFMFUSHWDVRU7JBHMXEATUXPI2HANCNFSM5LR5XCXQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

GladosBlueWallet commented:
Wake the fuck up samurai, we have PRs to merge

![image](https://user-images.githubusercontent.com/1913337/127044132-147fa451-6dc1-454f-907a-9292033047f2.png)

[all PRs for @Overtorment] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/Overtorment

Overtorment commented on class/wallets/multisig-hd-wallet.js:
```diff
@@ -640,11 +658,13 @@ export class MultisigHDWallet extends AbstractHDElectrumWallet {
-    for (let c = 0; c < this._cosigners.length; c++) {
-      const cosigner = this._cosigners[c];
-      const path = this._getDerivationPathByAddressWithCustomPath(input.address, this._cosignersCustomPaths[c] || this._derivationPath);
+    for (const [cosignerIndex, cosigner] of this._cosigners.entries()) {
```
smart

GladosBlueWallet commented:
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!
